tensorflow/python/kernel_tests/math_ops/batch_matmul_op_test.py

```diff
@@ -29,9 +30,11 @@
 
 
 def GetRandomNormalInput(shape, dtype):
-  # float16 has limited range so we reduce the variance of the scalars.
-  scale = 10.0 if dtype != np.float16 else 0.1
-  loc = -10.0 if dtype != np.float16 else 0.1
+  # float16 or bfloat16 has limited range so we reduce the variance of the
+  # scalars.
```

Comment:
Oh, yes, I think what I meant to do was to limit the scale of the inputs, otherwise, I saw the bfloat16 will output like:
```
Not equal to tolerance rtol=19.8425, atol=19.8425
Mismatched value: a is different from b.
not close where = (array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), array([ 60,  62,  65,  66,  67,  68,  69, 120, 121, 122, 123, 124, 125,
       126, 127, 128]))
not close lhs = [-25.    -25.    -25.    -25.    -25.    -25.    -25.    -22.375 -22.375
 -22.375 -22.375 -22.375 -22.375 -22.375 -22.375 -22.375]
not close rhs = [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
not close dif = [25.    25.    25.    25.    25.    25.    25.    22.375 22.375 22.375
 22.375 22.375 22.375 22.375 22.375 22.375]
not close tol = [19.842514 19.842514 19.842514 19.842514 19.842514 19.842514 19.842514
 19.842514 19.842514 19.842514 19.842514 19.842514 19.842514 19.842514
 19.842514 19.842514]
dtype = bfloat16, shape = (1, 210)
Mismatched elements: 16 / 210 (7.62%)
Max absolute difference: 106.625
Max relative difference: 0.82655036
 x: array([[-46.75 , -46.75 , -46.75 , -46.75 , -46.75 , -46.75 , -46.75 ,
        -46.75 , -46.75 , -46.75 , -51.   , -51.   , -51.   , -51.   ,
        -51.   , -51.   , -51.   , -51.   , -51.   , -51.   , -52.5  ,...
 y: array([[-129., -129., -129., -129., -129., -129., -129., -129., -129.,
        -129., -129., -129., -129., -129., -129., -129., -129., -129.,
        -129., -129., -129., -129., -129., -129., -129., -129., -129.,...

```

Reply:
But why would a large scale result in such incorrect outputs? It shouldn't make a difference whether the scale is .1 or 10.

Reply:
Yes, I think I found the root cause. It is because the loss function used is the reduce_sum, which would generate high error over these bfloat16 values ranging 10.0. So, I cast the results to float32 before the reduce_sum and keep the inputs as what they were. New commit has been push. PTAL. @reedwm